### ISSUE TITLE
Linking System-wide RocksDB reduces performance by 100x

### ISSUE THREAD

**Issue by preston-evans98** [Sovereign-Labs/sovereign-sdk#857]
## Summary
Linking the rust RocksDB crate against a system-wide RocksDB lib reduces performance by 100x.

## Steps to reproduce
Running `cargo bench` against my system-wide RocksDB installation shows ~50tps. 

```
[env]
ROCKSDB_LIB_DIR = "/usr/local/lib/"
SNAPPY_LIB_DIR = "/usr/local/lib/"
LZ4_LIB_DIR = "/usr/local/lib/"
```

If I delete the env from  `~/.cargo/config.toml` and recompile, performance returns to the expected number (~5k tps).

**Comment by DSHIVAAY-23** [comment]
@preston-evans98 i cannot find the exact file location and 
[env]
ROCKSDB_LIB_DIR = "/usr/local/lib/"
SNAPPY_LIB_DIR = "/usr/local/lib/"
LZ4_LIB_DIR = "/usr/local/lib/"
in this repo..
i think while we run cargo commands it makes a demo_data . [storage]
 The path to the rollup's data directory. Paths that do not begin with `/` are interpreted as relative paths.
path = "benches/demo_data"
And  can you fix it by your side that it will be default configuration for benches ?

**Comment by preston-evans98** [comment]
Hey @DSHIVAAY-23, we're investigating whether there are any changes we can make on our end to fix this - unfortunately it looks like its a RocksDB issue so we have limited control. 

For context, these configs live in `~/.cargo/config.toml`, not inside the repo - and they'll only exist there if you've set them up, probably while working on Substrate or another project that also uses RocksDB. These configs instruct the rust RocksDB wrapper crate not to recompile RocksDB from scratch - but it looks like performance is very sensitive to the exact compilation process that the Rust crate uses. The path to the data directory shouldn't matter at all though.

**Comment by DSHIVAAY-23** [comment]
Hey  @preston-evans98  can we change the no of block from 100 to 10 and lower the no of transactions locally. i think it will work on system without changing in rocksdb config .

**Comment by preston-evans98** [comment]
Yeah, you should be able to change that locally with no changes to the code. Both variables are set in the MAKEFILE - just update the values there and you should be good!

**Comment by DSHIVAAY-23** [comment]
Hey @preston-evans98  I want to know that if we implement decentralised-sequencing in future on zk-rollups. then we can get rid of on-chain proof verification on layer 1. we can verify that proof on L2 rather then sending that big data to L1?

**Comment by preston-evans98** [comment]
Hi @DSHIVAAY-23. Even today, you don't have to *verify* the proof on the L1. But you always have to *post* the proof on L1 if you want to be able to reward the prover. (Otherwise, you won't be able to come to consensus about who made the proof)

**Comment by DSHIVAAY-23** [comment]
![image](https://github.com/Sovereign-Labs/sovereign-sdk/assets/93729643/dde72d8b-e40d-4690-9488-9beb50c29290)
@preston-evans98 its unning now but for now we are getting low-tps  then expected idk its low network or its the real tps we gonna see.

**Comment by preston-evans98** [comment]
@DSHIVAAY-23, that's about an order of magnitude less than we're seeing in benchmarks. Can you share your setup (OS, CPU, and disk)?

**Comment by DSHIVAAY-23** [comment]
@preston-evans98 Ubuntu 22.04.3 ,11th gen intel i5 , Disk- 16GB and 512GB config..
